src/app/core/relations/relation.js

```diff
@@ -1,98 +1,154 @@
 const {base, inherit} = require('core/utils/utils');
 const G3WObject = require('core/g3wobject');
 
+/**
+ * Create a QGIS Relation Property: one (1-N) and many to many (N-M)
+ * 
+ * @class 
+ * @see https://docs.qgis.org/3.22/en/docs/user_manual/introduction/qgis_configuration.html?highlight=relation#relations-properties
+ */
 function Relation(config={}) {
+
   const uniqueSuffix = Date.now();
-  const id = config.id || `id_${uniqueSuffix}`;
-  const name = config.name || `name_${uniqueSuffix}`;
-  const origname = config.origname || `origname_${uniqueSuffix}`;
+
+  /**
+   * Store relation state
+   * @type {object}
+   */
   this.state = {
-    id,
-    name,
-    origname,
+    id: config.id || `id_${uniqueSuffix}`,
+    name: config.name || `name_${uniqueSuffix}`,
+    origname: config.origname || `origname_${uniqueSuffix}`,
     father: config.referencedLayer,
     child: config.referencingLayer,
-    fatherField: config.fieldRef.referencedField,
-    childField: config.fieldRef.referencingField,
-    type: config.type,
-    loading: false
+    fatherFields: config.fieldsRef.map(fieldRef => fieldRef.referencedField), // father Fields
+    childFields: config.fieldsRef.map(fieldRef => fieldRef.referencingField), // child fields
+    type: config.type
   };
+
   base(this);
+
 }
 
 inherit(Relation, G3WObject);
 
 const proto = Relation.prototype;
 
+/**
+ * @returns {string} relation identifier
+ */
 proto.getId = function() {
   return this.state.id;
 };
 
+/**
+ * @param {string} id relation identifier
+ */
 proto.setId = function(id) {
   this.state.id = id;
 };
 
+/**
+ * @returns {string} relation name
+ */
 proto.getName = function() {
   return this.state.name;
 };
 
+/**
+ * @param {string} name relation name
+ */
 proto.setName = function(name) {
   this.state.name = name;
 };
 
+/**
+ * @TODO check if this is currently a deprecated feature
+ * 
+ * @returns {string} relation title
+ */
 proto.getTitle = function() {
   return this.state.title;
 };
 
+/**
+ * @TODO check if this is currently a deprecated feature
+ * 
+ * @param {string} title relation title
+ * @returns {void}
+ */
 proto.setTitle = function(title) {
   return this.state.title = title;
```

Comment:
```js
this.state.title
```

it is not defined in the class's constructor, check if this is a refactoring oversight